cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMapping.java

```diff
@@ -79,63 +85,10 @@ void cacheRegulatingControls(PropertyBag p) {
-                    adder.setRegulationMode(StaticVarCompensator.RegulationMode.OFF);
-                    return;
-                }
-                if (context.terminalMapping().areAssociated(p.getId(TERMINAL), control.topologicalNode)) {
-                    setRegulatingControl(control, adder, idEq);
-                } else {
-                    context.pending(String.format("Remote control for static var compensator %s replaced by voltage local control at nominal voltage", idEq),
-                            "IIDM model does not support remote control for static var compensators");
-                    setDefaultRegulatingControl(p, adder, idEq);
-                }
-            } else {
-                context.missing(String.format(REGULATING_CONTROL_REF, p.getId(REGULATING_CONTROL)));
-                setDefaultRegulatingControl(p, adder, idEq);
-            }
-        } else {
-            setDefaultRegulatingControl(p, adder, idEq);
-        }
-    }
-
-    private void setDefaultRegulatingControl(PropertyBag p, StaticVarCompensatorAdder adder, String idEq) {
-        if (p.getId("controlMode").toLowerCase().endsWith(VOLTAGE)) {
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.VOLTAGE)
-                    .setVoltageSetPoint(p.asDouble("voltageSetPoint"));
-        } else if (p.getId("controlMode").toLowerCase().endsWith("reactivepower")) {
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.REACTIVE_POWER)
-                    .setReactivePowerSetPoint(p.asDouble("q"));
-        } else {
-            context.fixed("SVCControlMode", String.format("Invalid control mode for static var compensator %s. Regulating control is disabled", idEq));
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.OFF);
-        }
-    }
-
-    private void setRegulatingControl(RegulatingControl control, StaticVarCompensatorAdder adder, String idEq) {
-        if (control.mode.toLowerCase().endsWith(VOLTAGE)) {
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.VOLTAGE)
-                    .setVoltageSetPoint(control.targetValue);
-        } else if (control.mode.toLowerCase().endsWith("reactivepower")) {
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.REACTIVE_POWER)
-                    .setReactivePowerSetPoint(control.targetValue);
-        } else {
-            context.fixed(control.mode, String.format("Invalid control mode for static var compensator %s. Regulating control is disabled", idEq));
-            adder.setRegulationMode(StaticVarCompensator.RegulationMode.OFF);
-        }
-    }
-
     public void setAllRegulatingControls(Network network) {
         regulatingControlMappingForGenerators.applyRegulatingControls(network);
         regulatingControlMappingForTransformers.applyTapChangersRegulatingControl(network);
+        regulatingControlMappingForStaticVarCompensators.applyRegulatingControls(network);
```

Comment:
A general remark: after applying your changes, could you clear the map for generators, transformers and svc? We only clear the map of cached regulating controls...

Reply:
Why we should clear explicitly these temporal objects? the `RegulatingControlMapping` lives in the `Conversion.Context`, that will get out of scope after the import/load of CGMES data is finished. The garbage collector will take care of all the temporal objects that were used during the conversion.

Garbage collectors these days seem to be very efficient and maybe worrying about clearing and setting references to null as a general strategy is not really needed.